packages/cart/src/services/cart-module.ts

```diff
@@ -1100,4 +1069,352 @@ export default class CartModuleService implements ICartModuleService {
 
     await this.shippingMethodAdjustmentService_.delete(ids, sharedContext)
   }
+
+  @InjectManager("baseRepository_")
+  async listLineItemTaxLines(
+    filters: CartTypes.FilterableLineItemTaxLineProps = {},
+    config: FindConfig<CartTypes.LineItemTaxLineDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const taxLines = await this.lineItemTaxLineService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<CartTypes.LineItemTaxLineDTO[]>(
+      taxLines,
+      {
+        populate: true,
+      }
+    )
+  }
+
+  addLineItemTaxLines(
+    taxLines: CartTypes.CreateLineItemTaxLineDTO[]
+  ): Promise<CartTypes.LineItemTaxLineDTO[]>
+  addLineItemTaxLines(
+    taxLine: CartTypes.CreateLineItemTaxLineDTO
+  ): Promise<CartTypes.LineItemTaxLineDTO>
+  addLineItemTaxLines(
+    cartId: string,
+    taxLines:
+      | CartTypes.CreateLineItemTaxLineDTO[]
+      | CartTypes.CreateShippingMethodTaxLineDTO,
+    sharedContext?: Context
+  ): Promise<CartTypes.LineItemTaxLineDTO[]>
+
+  @InjectTransactionManager("baseRepository_")
+  async addLineItemTaxLines(
+    cartIdOrData:
+      | string
+      | CartTypes.CreateLineItemTaxLineDTO[]
+      | CartTypes.CreateLineItemTaxLineDTO,
+    taxLines?:
+      | CartTypes.CreateLineItemTaxLineDTO[]
+      | CartTypes.CreateLineItemTaxLineDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<CartTypes.LineItemTaxLineDTO[] | CartTypes.LineItemTaxLineDTO> {
+    let addedTaxLines: LineItemTaxLine[] = []
+    if (isString(cartIdOrData)) {
+      // existence check
+      await this.retrieve(cartIdOrData, { select: ["id"] }, sharedContext)
```

Comment:
question: this will not return any error if it does not exists, do we miss a check?

Reply:
Not sure I understand. `this.retrieve` will throw in case the cart does not exist, if that's what you are asking

Reply:
yes you are right, I forgot and couldn't remember if we added the throw, discard those comments about it 👍